os_net_config/impl_nmstate.py

```diff
@@ -1833,6 +1840,34 @@ def parse_ovs_extra(self, ovs_extras, name, data):
             for cmd_map in cfg_val_pair:
                 self._ovs_extra_cfg_val(ovs_extra_cmd, cmd_map, data)
 
+    def parse_ovs_extra_for_iface(self, ovs_extras, iface_name, data):
+        """Parse ovs extra for interface
+
+        :param ovs_extras: given ovs extra as string
+        :param iface_name: Interface name
+        :param data: The interface data that will be modified after
+            the parsing
+        """
+        iface_cfg = [{'config': r'^external-ids:[\w+]',
+                      'sub_tree': [OvsDB.KEY, OvsDB.EXTERNAL_IDS],
+                      'nm_config': None,
+                      'nm_config_regex': r'^external-ids:(.+?)=',
+                      'value_pattern': r'^external-ids:.*=(.+?)$'},
+                     {'config': r'^external_ids:[\w+]',
+                      'sub_tree': [OvsDB.KEY, OvsDB.EXTERNAL_IDS],
+                      'nm_config': None,
+                      'nm_config_regex': r'^external_ids:(.+?)=',
+                      'value_pattern': r'^external_ids:.*=(.+?)$'}
+                     ]
```

Comment:
I'm not sure I understand why this iface_cfg is a list of these two identical dicts. Maybe add a docstring if you do another pass.